lib/ocso_scraper/crawler.rb

```diff
@@ -0,0 +1,79 @@
+require 'httparty'
+require 'uri'
+require 'net/http'
+require 'openssl'
+
+module OcsoScraper
+  class Crawler
+    def self.perform
+      warrants = []
+      ('A'..'Z').each do |first_name_first_letter|
+        ('A'..'Z').each do |last_name_first_letter|
+          last_name_query = last_name_first_letter
+          next_last_name_query = nil
+          continue = true
+          while continue
+            results = request(first_name_first_letter, last_name_query)
+            page_warrants = PageParser.perform(results).sort_by! { |warrant| warrant[:last_name] }
+            warrants += page_warrants
+            if next_last_name_query.present?
+              last_name_query = next_last_name_query
+              next_last_name_query = nil
+            elsif page_warrants.count < page_limit
```

Comment:
if page_warrants.count is less than page limit then continue is false.?(little confused about this)